.mci.yml

```diff
@@ -1174,6 +1187,28 @@ buildvariants:
             - debian10-large
           - name: uninstall_check
 
```

Comment:
I suggest reverting the changes to the `buildvariants` section and adding this build variant:

```yaml
    # Add matrix for specification test requirement of mongocryptd:
    # "Drivers MUST run all tests with mongocryptd on at least one platform for all tested server versions (4.2+)."
    - matrix_name: "mongocryptd"
      matrix_spec:
        os: "ubuntu-1804"
        mongodb_version: ["4.2", "4.4", "5.0", "latest"]
      display_name: "${os} (MongoDB ${mongodb_version}) with mongocryptd"
      tasks:
          - name: compile_and_test_with_shared_libs
          - name: compile_and_test_with_shared_libs_replica_set
      expansions:
        use_mongocryptd: true
```

I suggest only testing `compile_and_test_with_shared_libs` and `compile_and_test_with_shared_libs_replica_set`. Other tasks (e.g. `*_extra_alignment` and `*_static_libs`) do not seem directly related to testing mongocryptd or crypt_shared.

I prefer not changing the build variants below. E.g. `ubuntu1604-debug-std-experimental` has a specific purpose unrelated to testing with mongocryptd or crypt_shared.

Reply:
Great! I added the "mongocryptd" test as specified above.

The reason for me disabling the crypt_shared tests for the Valgrind and std-experimental tasks is that for Valgrind, we're getting false positives when using the crypt_shared library. The ASAN tests are passing and not detecting any leaks, but Valgrind has [known issues when running optimized code](https://stackoverflow.com/questions/20809080/how-does-optimization-trip-valgrind-false-positives) (such as the crypt_shared library) and is erroring with false positive leaks:
```
[2023/04/17 15:25:18.138] All tests passed (6078 assertions in 211 test cases)
[2023/04/17 15:25:18.701] ==9264==
[2023/04/17 15:25:18.701] ==9264== HEAP SUMMARY:
[2023/04/17 15:25:18.701] ==9264==     in use at exit: 542,627 bytes in 6,516 blocks
[2023/04/17 15:25:18.701] ==9264==   total heap usage: 1,032,738 allocs, 1,026,222 frees, 1,167,109,914 bytes allocated
[2023/04/17 15:25:18.701] ==9264==
[2023/04/17 15:25:19.115] ==9264== LEAK SUMMARY:
[2023/04/17 15:25:19.115] ==9264==    definitely lost: 0 bytes in 0 blocks
[2023/04/17 15:25:19.115] ==9264==    indirectly lost: 20 bytes in 2 blocks
[2023/04/17 15:25:19.115] ==9264==      possibly lost: 0 bytes in 0 blocks
[2023/04/17 15:25:19.115] ==9264==    still reachable: 542,415 bytes in 6,513 blocks
[2023/04/17 15:25:19.115] ==9264==         suppressed: 192 bytes in 1 blocks
[2023/04/17 15:25:19.115] ==9264== Reachable blocks (those to which a pointer was found) are not shown.
[2023/04/17 15:25:19.115] ==9264== To see them, rerun with: --leak-check=full --show-leak-kinds=all
[2023/04/17 15:25:19.115] ==9264==
[2023/04/17 15:25:19.115] ==9264== For counts of detected and suppressed errors, rerun with: -v
[2023/04/17 15:25:19.115] ==9264== ERROR SUMMARY: 25 errors from 2 contexts (suppressed: 1 from 1)
[2023/04/17 15:25:19.158] Command 'shell.exec' in function 'test' failed: shell script encountered problem: exit code 1.
```

For the case of std-experimental, there are failures to open the crypt_shared library:
```
[2023/04/17 15:20:53.592] due to unexpected exception with message:
[2023/04/17 15:20:53.592]   A crypt_shared override path was specified [], but we failed to open a
[2023/04/17 15:20:53.592]   dynamic library at that location: generic server error
```

Reply:
It appears that crypt_shared is not available for Ubuntu 16.04, which was the cause of this issue. I updated the code to use mongocryptd on Ubuntu 16.04.